staging/kos/pkg/controllers/connectionagent/connectionagent.go

```diff
@@ -0,0 +1,1091 @@
+			remAttState := ca.getAttState(remAttNSN)
+			if remAttState == nil {
+				remAttState = &attachmentState{
+					vnStateVNI: ifcVNI,
+				}
+				ca.setAttState(remAttNSN, remAttState)
+			}
+			ca.updateVNState(remAttState, ifcVNI, remAttNSN, ifcOwnerAtt.Spec.Node)
+			oldIfc, oldIfcExists := remAttState.ifc, remAttState.ifcIsValid
+			remAttState.vnStateVNI, remAttState.ifc, remAttState.ifcIsValid = ifcVNI, aPreExistingRemoteIfc, true
+			glog.V(3).Infof("Init: matched interface %#+v with remote attachment %#+v", aPreExistingRemoteIfc, ifcOwnerAtt)
+			if oldIfcExists {
+				aPreExistingRemoteIfc = oldIfc
+			} else {
+				continue
+			}
+		}
+
+		// If we're here either there's no remote attachments cache associated with the interface vni (because there are
+		// no local attachments with that vni), or no remote attachment owning the interface was found, or the attachment
+		// owning the interface already has one. For all such cases we need to delete the interface.
+		for i, err := 1, ca.netFabric.DeleteRemoteIfc(aPreExistingRemoteIfc); err != nil; i++ {
+			glog.V(3).Infof("Init: deletion of orphan remote interface %#+v failed: %s. Attempt nbr. %d",
+				aPreExistingRemoteIfc,
+				err.Error(),
+				i)
+			time.Sleep(netFabricRetryPeriod)
+		}
+		glog.V(3).Infof("Init: deleted orphan local interface %#+v", aPreExistingRemoteIfc)
+	}
+
+	return nil
+}
+
+func (ca *ConnectionAgent) processQueue() {
+	for {
+		item, stop := ca.queue.Get()
+		if stop {
+			return
+		}
+		attNSN := item.(k8stypes.NamespacedName)
+		ca.processQueueItem(attNSN)
+	}
+}
+
+func (ca *ConnectionAgent) processQueueItem(attNSN k8stypes.NamespacedName) {
+	defer ca.queue.Done(attNSN)
+	err := ca.processNetworkAttachment(attNSN)
+	requeues := ca.queue.NumRequeues(attNSN)
+	if err != nil {
+		// If we're here there's been an error: either the attachment current state was
+		// ambiguous (e.g. more than one vni), or there's been a problem while processing
+		// it (e.g. Interface creation failed). We requeue the attachment reference so that
+		// it can be processed again and hopefully next time there will be no errors.
+		glog.Warningf("Failed processing NetworkAttachment %s, requeuing (%d earlier requeues): %s",
+			attNSN,
+			requeues,
+			err.Error())
+		ca.queue.AddRateLimited(attNSN)
+		return
+	}
+	glog.V(4).Infof("Finished NetworkAttachment %s with %d requeues", attNSN, requeues)
+	ca.queue.Forget(attNSN)
+}
+
+func (ca *ConnectionAgent) processNetworkAttachment(attNSN k8stypes.NamespacedName) error {
+	att, deleted, err := ca.getAttachment(attNSN)
+	if att != nil {
+		// If we are here the attachment exists and it's current state is univocal
+		err = ca.processExistingAtt(att)
+	} else if deleted {
+		// If we are here the attachment has been deleted
+		err = ca.processDeletedAtt(attNSN)
+	}
+	return err
+}
+
+// getAttachment attempts to determine the univocal version of the NetworkAttachment with namespaced name attNSN. If it succeeds it
```

Comment:
Here is what this looks like to me.  Why do we want such wide lines?

![screen shot 2018-12-19 at 1 02 38 pm](https://user-images.githubusercontent.com/14296719/50238855-7bdf9380-038e-11e9-99f8-3a0ed85f4985.png)

Reply:
Are you referring to comments, code or both?

Reply:
I was mainly thinking of the comments, which were consistently broken at long line lengths.  golang code style de-emphasizes breaking code lines, although I sometimes do that anyway for the sake of readability.